app/src/main/java/org/oppia/android/app/hintsandsolution/HintsViewModel.kt

```diff
@@ -98,7 +99,10 @@ class HintsViewModel @Inject constructor(
   fun computeHintListDropDownIconContentDescription(): String {
     return resourceHandler.getStringInLocaleWithWrapping(
       R.string.show_hide_hint_list,
-      hintsAndSolutionSummary.get() ?: DEFAULT_HINT_AND_SOLUTION_SUMMARY
+      HtmlCompat.fromHtml(
```

Comment:
We should only be using our own HtmlParser when parsing HTML to ensure consistency and support for custom tags.

I think this will require doing something similar to the parseOppiaHtml call in HintsAndSolutionDialogFragmentPresenter, though you'll need to update parseOppiaHtml itself to support not taking a TextView (for cases when we just want to parse into spannables without tweaking the textview itself).

Reply:
I have updated parseOppiaHtml and related code.